api/mapping/services_rules.py

```diff
@@ -841,3 +843,101 @@ def remove_mapping_rules(request, scan_report_id):
+                    source_ids = (
+                        MappingRule.objects.filter(Q(concept__concept=desc))
+                        .exclude(
+                            Q(omop_field__field__icontains="person_id")
+                            | Q(omop_field__field__icontains="datetime")
+                            | Q(omop_field__field__icontains=("source"))
+                        )
+                        .values(
+                            "source_field__id",
+                            "source_field__name",
+                            "source_field__scan_report_table__id",
+                            "source_field__scan_report_table__name",
+                            "concept__content_type",
+                        )
+                    ).distinct()
+                    descendant_list.append(
+                        {
+                            "d_id": desc,
+                            "d_name": concept.concept_name,
+                            "source": source_ids,
+                        }
+                    )
+            for ancestor in ancestors:
+
+                anc = ancestor.ancestor_concept_id
+
+                if (anc in all_mapping_rules) & (anc != rule):
+                    concept = Concept.objects.get(concept_id=anc)
+                    rule_name = Concept.objects.get(concept_id=rule).concept_name
+                    source_ids = (
+                        MappingRule.objects.filter(Q(concept__concept=anc))
+                        .exclude(
+                            Q(omop_field__field__icontains="person_id")
+                            | Q(omop_field__field__icontains="datetime")
+                            | Q(omop_field__field__icontains="source")
+                        )
+                        .values(
+                            "source_field__id",
+                            "source_field__name",
+                            "source_field__scan_report_table__id",
+                            "source_field__scan_report_table__name",
+                            "concept__content_type",
+                        )
+                        .distinct()
+                    )
+
+                    ancestors_list.append(
+                        {
+                            "a_id": anc,
+                            "a_name": concept.concept_name,
+                            "source": source_ids,
+                        }
+                    )
+            data.append(
+                {
+                    "rule_id": rule,
+                    "rule_name": rule_name,
+                    "anc_desc": [
+                        {
+                            "descendants": descendant_list,
+                            "ancestors": ancestors_list,
+                        }
+                    ],
+                }
+            )
+            descendant_list = []
+            ancestors_list = []
+        except:
```

Comment:
If this really does need to be in a `try`/`except` loop, could we put something in here to explain what went wrong please?

Reply:
I agree that we shouldn't have a bare except here. If nothing can go wrong, then remove the try/except. If something can go wrong, then handle each case specifically. Otherwise, any error in the above will silently be glossed over, which will be enormously difficult to debug.

Reply:
Changed it to show this in the terminal if there are no matches in other Scan Reports:
"No ancestors or descendants of <conceptID> appear in any other Scan Reports"

Also added a message in the pop-up modal to say "No ancestors or descendants of these mappings appear in any other Scan Reports" instead of just being empty